code/modules/library/library_computer.dm

```diff
@@ -0,0 +1,577 @@
+				if("edit_search_author")
+					ui_modal_input(src, id, "Please input the new author:", null, arguments, user_data.search_author)
+				if("edit_search_ratingmax")
+					ui_modal_input(src, id, "Please input the new upper rating bound:", null, arguments,  user_data.search_rating["max"])
+				if("edit_search_ratingmin")
+					ui_modal_input(src, id, "Please input the new lower rating bound:", null, arguments,  user_data.search_rating["min"])
+				//book uploader inputs
+				if("edit_selected_title")
+					ui_modal_input(src, id, "Please input the new title:", null, arguments, user_data.selected_book.title)
+				if("edit_selected_author")
+					ui_modal_input(src, id, "Please input the new author:", null, arguments, user_data.selected_book.author)
+				if("edit_selected_summary")
+					ui_modal_input(src, id, "Please input the new summary:", null, arguments, user_data.selected_book.summary)
+				//book list buttons
+				if("expand_info")
+					var/datum/programmatic_book/PB = GLOB.library_catalog.get_programmatic_book_by_id(arguments["bookid"])
+					if(PB)
+						ui_modal_message(src, id, "", arguments = list(
+							"isProgrammatic" = TRUE,
+							"title" = PB.title,
+							"author" = PB.author,
+							"summary" = PB.summary ? PB.summary : "No Summary Provided",
+							"rating" = "N for Nanotrasen",
+						))
+
+						return //If we've succesfully opened the modal for our programmatic book, we don't need to do more logic
+					var/datum/cachedbook/CB = GLOB.library_catalog.get_book_by_id(arguments["bookid"])
+					if(CB)
+						var/category_names = list()
+						for(var/datum/library_category/category in CB.categories)
+							category_names += category.description
+						user_data.selected_report = null
+						user_data.selected_rating = 0
+
+						ui_modal_message(src, id, "", arguments = list(
+							"isProgrammatic" = FALSE,
+							"id" = CB.id,
+							"ckey" = CB.ckey,
+							"title" = CB.title,
+							"author" = CB.author,
+							"summary" = CB.summary ? CB.summary : "No Summary Provided",
+							"rating" = CB.rating ? CB.rating : "No Rating Provided",
+							"categories" = category_names,
+						))
+				if("report_book")
+					var/datum/cachedbook/CB = GLOB.library_catalog.get_book_by_id(arguments["bookid"])
+					ui_modal_message(src, id, "", arguments = list(
+						id = CB.id,
+						title = CB.title,
+						ckey = CB.ckey,
+					))
+				if("rate_info")
+					var/datum/cachedbook/CB = GLOB.library_catalog.get_book_by_id(arguments["bookid"])
+					var/list/book_ratings = GLOB.library_catalog.get_book_ratings(arguments["bookid"])
+					ui_modal_message(src, id, "", arguments = list(
+						"id" = CB.id,
+						"title" = CB.title,
+						"author" = CB.author,
+						"ckey" = CB.ckey,
+						"current_rating" = length(book_ratings) ? book_ratings[1] : 0,
+						"total_ratings" = length(book_ratings) ? length(book_ratings[2]) : 0,
+					))
+				else
+					return FALSE
+		if(UI_MODAL_ANSWER)
+			var/answer = params["answer"]
```

Comment:
Prob want to sanitize the answer here as well to avoid xss attacks etc.

Reply:
done 👍 
I didn't even think about xss attacks here